app/gui/view/project-view-top-bar/src/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,457 @@
+        }
+    }
+}
+
+
+
+// ========================
+// === RelativePosition ===
+// ========================
+
+/// The position of this breadcrumb relative to the selected breadcrumb. We use this to determine
+/// the color.
+#[derive(Debug, Clone, Copy)]
+enum RelativePosition {
+    Left,
+    Right,
+}
+
+
+
+// ==================
+// === Animations ===
+// ==================
+
+/// ProjectName's animations handlers.
+#[derive(Debug, Clone, CloneRef)]
+pub struct Animations {
+    color:           Animation<color::Rgba>,
+    separator_color: Animation<color::Rgba>,
+}
+
+impl Animations {
+    /// Constructor.
+    pub fn new(network: &frp::Network) -> Self {
+        let color = Animation::new(network);
+        let separator_color = Animation::new(network);
+        Self { color, separator_color }
+    }
+}
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+/// Breadcrumb frp network inputs.
+#[derive(Debug, Clone, CloneRef)]
+pub struct FrpInputs {
+    /// Select the breadcrumb, triggering the selection animation.
+    pub select:   frp::Source,
+    /// Select the breadcrumb, triggering the deselection animation, using the (self,new)
+    /// breadcrumb indices to determine if the breadcrumb is on the left or on the right of the
+    /// newly selected breadcrumb.
+    pub deselect: frp::Source<(usize, usize)>,
+}
+
+impl FrpInputs {
+    /// Constructor.
+    pub fn new(network: &frp::Network) -> Self {
+        frp::extend! {network
+            select   <- source();
+            deselect <- source();
+        }
+        Self { select, deselect }
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+/// Breadcrumb frp network outputs.
+#[derive(Debug, Clone, CloneRef)]
+pub struct FrpOutputs {
+    /// Signalizes that the breadcrumb was clicked.
+    pub clicked:     frp::Source,
+    /// Signalizes that the breadcrumb's size changed.
+    pub size:        frp::Source<Vector2<f32>>,
+    /// Signalizes the breadcrumb's selection state.
+    pub selected:    frp::Source<bool>,
+    /// Used to check if the breadcrumb is selected.
+    pub is_selected: frp::Sampler<bool>,
+    /// Width of the breadcrumb.
+    pub width:       frp::Source<f32>,
+}
+
+impl FrpOutputs {
+    /// Constructor.
+    pub fn new(network: &frp::Network) -> Self {
+        frp::extend! { network
+            clicked     <- source();
+            size        <- source();
+            selected    <- source();
+            is_selected <- selected.sampler();
+            width <- source();
+        }
+        Self { clicked, size, selected, is_selected, width }
+    }
+}
+
+
+
+// ===========
+// === Frp ===
+// ===========
+
+/// A breadcrumb frp structure with its endpoints and network representation.
+#[derive(Debug, Clone, CloneRef)]
+#[allow(missing_docs)]
+pub struct Frp {
+    pub inputs:  FrpInputs,
+    pub outputs: FrpOutputs,
+    pub network: frp::Network,
+}
```

Comment:
I guess this is just an old code just moved, and that's why we define outputs in such unusual way?

Reply:
Yep, copy-pasted. And it's not easy to change it to `define_endpoints`, because `breadcrumbs` (parent module) uses them in unusual way.